lib/reporter.js

```diff
@@ -0,0 +1,157 @@
+  let errorsCounterName = metrics.deltaCounterName(config.standardLambdaMetrics.errorsCounter);
+  registry.addTaggedMetric(errorsCounterName, errorsCounter);
+
+	// Register cold start counter.
+  coldStartEventCounter = new metrics.Counter();
+  registry.addTaggedMetric(config.standardLambdaMetrics.coldStartEventCounter, coldStartEventCounter);
+
+  coldStartsCounter = new metrics.Counter();
+  let coldStartsCounterName = metrics.deltaCounterName(config.standardLambdaMetrics.coldStartsCounter);
+  registry.addTaggedMetric(coldStartsCounterName, coldStartsCounter);
+
+  // Register duration as Counter.
+  durationValue = new metrics.Counter();
+  registry.addTaggedMetric(config.standardLambdaMetrics.durationValue, durationValue);
+
+  if(config.isColdStart){
+    //Update cold start counter.
+    incrementColdStarts()
+    config.isColdStart = false;
+  }
+}
+
+// Function to increment or update the given metric with the given value.
+// This method should only be used to handle updating values of standardLambdaMetrics
+// defined in ./config.js
+const incrementMetric = function (metricName, value) {
+  if (!config.isReportStandardMetrics) {
+    return;
+  }
+  switch (metricName) {
+    case config.standardLambdaMetrics.invocationsCounter:
+        invocationsCounter.inc(value);
+        break;
+    case config.standardLambdaMetrics.invocationEventCounter:
+        invocationEventCounter.inc(value);
+        break;
+    case config.standardLambdaMetrics.errorsCounter:
+        errorsCounter.inc(value);
+        break;
+    case config.standardLambdaMetrics.errorEventCounter:
+        errorEventCounter.inc(value);
+        break;
+    case config.standardLambdaMetrics.coldStartsCounter:
+        coldStartsCounter.inc(value);
+        break;
+    case config.standardLambdaMetrics.coldStartEventCounter:
+        coldStartEventCounter.inc(value);
+        break;
+    case config.standardLambdaMetrics.durationValue:
+        durationValue.inc(value);
+        break;
+    default:
+         console.warn("No standard lambda metric matched.");
+  }
+}
+
+const incrementInvocations = function(){
+  incrementMetric(config.standardLambdaMetrics.invocationEventCounter, 1)
+  incrementMetric(config.standardLambdaMetrics.invocationsCounter, 1)
+}
+
+const incrementErrors = function(){
+  incrementMetric(config.standardLambdaMetrics.errorEventCounter, 1)
+  incrementMetric(config.standardLambdaMetrics.errorsCounter, 1)
+}
+
+const incrementColdStarts = function(){
+  incrementMetric(config.standardLambdaMetrics.coldStartEventCounter, 1)
+  incrementMetric(config.standardLambdaMetrics.coldStartsCounter, 1)
+}
+
+const updateDuration = function(value){
+  incrementMetric(config.standardLambdaMetrics.durationValue, value)
+}
+
+const getRegistry = () => {
+  return registry;
+}
+
+const reportMetrics = (context) =>  {
+  if(context != null){
+    let invokedFunctionArn = context.invokedFunctionArn
+    let splitArn = invokedFunctionArn.split(":")
```

Comment:
Please use destructuring assignment to improve readability:
let [foo, bar, baz, etc] = invokedFunctionArn.split(":")